src/Store/SyntaxHighlightingStoreConnector.re

```diff
@@ -16,158 +16,218 @@ module Ext = Oni_Extensions;
 module NativeSyntaxHighlights = Oni_Syntax.NativeSyntaxHighlights;
 module Protocol = Oni_Syntax.Protocol;
 
-let start =
-    (languageInfo: Ext.LanguageInfo.t, setup: Core.Setup.t, cli: Core.Cli.t) => {
-  let (stream, dispatch) = Isolinear.Stream.create();
-
-  if (!cli.shouldSyntaxHighlight) {
-    let updater = (state, _action) => (state, Isolinear.Effect.none);
-    (updater, stream);
-  } else {
-    let onHighlights = tokenUpdates => {
-      dispatch(Model.Actions.BufferSyntaxHighlights(tokenUpdates));
+type subscriptionParams('msg) = {
+  id: string,
+  languageInfo: Ext.LanguageInfo.t,
+  setup: Core.Setup.t,
+  onStart: Oni_Syntax_Client.t => 'msg,
+  onClose: unit => 'msg,
+  onHighlights: list(Oni_Syntax.Protocol.TokenUpdate.t) => 'msg,
+};
+
+module Subscription =
```

Comment:
This first step of refactoring migrates from using the `Isolinear.Stream.create` to an actual managed `Subscription`.

The next step would be to factor this out to a `Service`, including the effects.